selectors/views/threads.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {$ID} from 'mattermost-redux/types/utilities';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {getThreads} from 'mattermost-redux/selectors/entities/threads';
+import {Team} from 'mattermost-redux/types/teams';
+import {UserThread} from 'mattermost-redux/types/threads';
+
+import {GlobalState} from 'types/store';
+import {ViewsState} from 'types/store/views';
+
+export function getSelectedThreadIdInTeam(state: GlobalState) {
+    return state.views.threads.selectedThreadIdInTeam;
+}
+
+export const getSelectedThreadIdInCurrentTeam: (state: GlobalState) => ViewsState['threads']['selectedThreadIdInTeam'][$ID<Team>] = createSelector(
+    getCurrentTeamId,
+    getSelectedThreadIdInTeam,
+    (
+        currentTeamId,
+        selectedThreadIdInTeam,
+    ) => {
+        return selectedThreadIdInTeam?.[currentTeamId] ?? null;
+    },
+);
+
+export const getSelectedThreadInCurrentTeam: (state: GlobalState) => UserThread | null = createSelector(
+    getCurrentTeamId,
+    getSelectedThreadIdInTeam,
+    getThreads,
+    (
+        currentTeamId,
+        selectedThreadIdInTeam,
+        threads,
+    ) => {
+        const threadId = selectedThreadIdInTeam?.[currentTeamId];
+        if (!threadId) {
```

Comment:
Redundant check given below?

Reply:
Looks like a bad merge, will clean it up